src/components/floating-emojis/FloatingEmoji.js

```diff
@@ -24,73 +26,82 @@ const FloatingEmoji = ({
   top,
   wiggleFactor,
 }) => {
-  const { opacity, rotate, scale, translateX, translateY } = useMemoOne(() => {
-    const animation = timing({ duration, easing: EasingNode.elastic() });
-    const progress = interpolate(animation, {
-      inputRange: [0, 1],
-      outputRange: [0, distance],
+  const animation = useSharedValue(0);
+
+  useLayoutEffect(() => {
+    animation.value = withTiming(1, {
+      duration,
+      easing: Easing.linear,
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const progress = interpolate(animation.value, [0, 1], [0, distance]);
+
+    const opacity = interpolate(
+      progress,
+      [0, distance * (opacityThreshold ?? 0.5), distance - size],
+      [1, fadeOut ? 0.89 : 1, fadeOut ? 0 : 1]
+    );
+
+    const rotate =
+      interpolate(
+        progress,
+        [0, distance / 4, distance / 3, distance / 2, distance],
+        [0, -2, 0, 2, 0]
+      ) + 'deg';
+
+    const scale = interpolate(
+      progress,
+      [0, 15, 30, 50, distance],
+      [0, 1.2, 1.1, 1, scaleTo]
+    );
+
+    const everyThirdEmojiMultiplier = index % 3 === 0 ? 3 : 2;
+    const everySecondEmojiMultiplier = index % 2 === 0 ? -1 : 1;
+    const translateXComponentA =
+      animation.value *
+      size *
+      everySecondEmojiMultiplier *
+      everyThirdEmojiMultiplier;
+
+    const wiggleMultiplierA = Math.sin(progress * (distance / (350 / 15)));
```

Comment:
I think I would still leave the comment here cause we still don't WTF are those numbers!!

cc-ing @mikedemarais although I'm 99.9% sure it's pointless

Reply:
```suggestion
    const wiggleMultiplierA = Math.sin(progress * (distance / 23.3));
```
 xD